src/solver/conv_asm_implicit_gemm_gtc_bwd.cpp

```diff
@@ -233,22 +252,70 @@ static std::tuple<bool,        // is suitable kernel found
             bool is_gemm_not_empty = gemm_k > 0;
             if(is_gemm_not_empty)
             {
-                if(gemm_k % pConfig->gemm_k_per_block != 0)
+                if(gemm_k % cfg.gemm_k_per_block != 0)
                     gemm_k_valid = false;
             }
         }
         if(!gemm_k_valid)
             continue;
-        // if all valid, break out;
-        break;
+        return std::make_tuple(true,
+                               cfg.GetKernelName(),
+                               cfg.GetBlockSize(),
+                               integer_divide_ceil(gemm_m, cfg.gemm_m_per_block) *
+                                   integer_divide_ceil(gemm_n, cfg.gemm_n_per_block));
     }
-    if(pConfig != tunables.end())
+
+    // second try, try find if packed image size match
+    for(const auto& cfg : tunables)
```

Comment:
Is it possible to find a suitable tunable configuration by one iteration?

Reply:
I would want to, but currently if change into one iteration it's better to re-consider the whole process of find kernel logic. If based on currently design it is hard to achieve without introduce regression